File: leetcode/java/1475. Final Prices With a Special Discount in a Shop.java
class Solution {
    public int[] finalPrices(int[] prices) {
        final int[] ret = new int[prices.length];
        
        for(int i = 0; i < prices.length; i++) {
            ret[i] = prices[i];
                
            for(int j = i+1; j < prices.length; j++) {
                if(prices[i] >= prices[j]) {
                    ret[i] = prices[i] - prices[j];
                    break;
                }
            }
        }
        
        return ret;
    }
}

File: leetcode/kotlin/1475. Final Prices With a Special Discount in a Shop.kt
class Solution {
    fun finalPrices(prices: IntArray): IntArray {
        val ret = IntArray(prices.size) {0}
                
        for(i in 0 until prices.size) {
            
            ret[i] = prices[i]
            
            for(j in (i+1) until prices.size) {
                if(prices[j] <= prices[i]) {
                    ret[i] = prices[i] - prices[j]
                    break
                }
            }
        }
        
        return ret
    }
}
